### PR TITLE
#3189 Add an error in UpdateXML to prevent using update scripts with the same name

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non-regression tests were added (new model, new feature and bug fix)
- [ ] main documentation was updated (update of input/output file, 3rd party, model, repository organization, solver)
- [ ] example documentations were updated (new example in examples folder)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies the parameters or inputs/outputs of a model/solver: the corresponding xsl was added in util/xsl and platform DB were updated
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

File: util/updateXML/content/sources/Job/Jobs.py [hunk 1]
```diff
@@ -265,35 +267,77 @@ def __get_update_modules_filepath_list(self, all_files_in_all_scripts_dirs):
...
+        for update_filepath in all_update_filepaths_in_all_scripts_dirs:
+            module_version = self.__get_module_version(update_filepath)
+            if min_limit_version < tuple(map(int, module_version.split('.'))) < max_limit_version:
+                unsorted_update_modules_list.append(update_filepath)
+
+        if len(unsorted_update_modules_list) == 0:
+            print("Error : Patch files not found")
+            sys.exit(1)
+
+        sorted_update_modules_list = sorted(unsorted_update_modules_list, key=lambda update_filepath: os.path.basename(update_filepath))
```
Comment (rosiereflo) [inline util/updateXML/content/sources/Job/Jobs.py:284]: Why is this added here?

File: util/updateXML/content/sources/Job/Jobs.py [hunk 2]
```diff
@@ -265,35 +267,77 @@ def __get_update_modules_filepath_list(self, all_files_in_all_scripts_dirs):
...
-                if not re.match(r'^[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+$', module_version):
-                    invalid_update_files.append(update_file)
+                if not re.match(MODULE_VERSION_REGEX, module_version):
+                    invalid_update_files.append(file_name)
                     continue
-                min_limit_version = (self.__dynawo_origin[0], self.__dynawo_origin[1], self.__dynawo_origin[2]+1)
-                max_limit_version = (self.__dynawo_version[0], self.__dynawo_version[1], self.__dynawo_version[2]+1)
-                if min_limit_version < tuple(map(int, module_version.split('.'))) < max_limit_version:
-                    unsorted_update_modules_list.append(update_filepath)
+                all_update_filepaths_in_all_scripts_dirs.append(file_path)
```
Comment (rosiereflo) [inline util/updateXML/content/sources/Job/Jobs.py:312]: Can't we simply check that there is not duplicated basename in this list ?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

File: util/updateXML/content/sources/Job/Jobs.py (hunk 2 above)
Comment (barondim) [reply util/updateXML/content/sources/Job/Jobs.py:312]: I thought using a function to check there is no duplicates would make the code more modular.
Moreover, with the check_duplicate_basenames() we can print all the duplicates. If we just check there is no duplicates when we add a file path, we'll just check one basename at a time.

File: util/updateXML/content/sources/Job/Jobs.py [hunk 3]
```diff
@@ -274,6 +276,9 @@ def __get_update_modules_filepath_list(self, all_files_in_all_scripts_dirs):
             if update_file.startswith(main_update_filename_without_extension) and \
                     update_file.endswith(PYTHON_FILE_EXTENSION) and \
                     update_file != main_update_filename:
+                if any(update_file == update_filename for update_filename in all_update_filenames_in_all_scripts_dirs):
```
Comment (rosiereflo) [inline util/updateXML/content/sources/Job/Jobs.py:279]: if update_file in all_update_filenames_in_all_scripts_dirs ?

**Review by rosiereflo — CHANGES_REQUESTED** [review]

**Review by rosiereflo — APPROVED** [review]
